sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -108,8 +108,9 @@ object Cast {
     case (TimestampType, TimestampNTZType) => true
 
     case (StringType, _: CalendarIntervalType) => true
-    case (StringType, _: DayTimeIntervalType) => true
-    case (StringType, _: YearMonthIntervalType) => true
+    case (StringType, _: AnsiIntervalType) => true
+
+    case (_: AnsiIntervalType, _: IntegralType) => true
```

Comment:
Shall we allow `IntegralType => AnsiIntervalType` as well?

Reply:
The SQL standard allows this. I think it makes sense to support this too.